### ISSUE TITLE
ARIA-hidden labels should not have been always considered "hidden"

### ISSUE THREAD

**Issue by Vittly** [dequelabs/axe-core#3287]
Product: axe-core

Expectation: `hidden-explicit-label` should not consider `label[aria-hidden="true"]` truly hidden if it references to `input` via `for=<ID_REF>`

Actual: it considers `label[aria-hidden="true"]` [as "not visible"](https://github.com/dequelabs/axe-core/blob/f62f0cf18f7b69b247b0b6362cf1ae71ffbf3a1b/lib/commons/dom/is-visible.js#L154-L156), and it fails [hidden-explicit-label](https://github.com/dequelabs/axe-core/blob/f62f0cf18f7b69b247b0b6362cf1ae71ffbf3a1b/test/checks/label/hidden-explicit-label.js#L61-L71) check

Motivation: **4.3.1. Terminology** of [Accessible Name and Description Computation 1.1]((https://www.w3.org/TR/accname-1.1) 
) states:

> The text alternative for a given element is computed as follows ... 2. ... A. If the current node is hidden and is not directly referenced by aria-labelledby or aria-describedby, nor directly referenced by a native host language text alternative element (e.g. label in HTML)

That means that if `label[aria-hidden="true"]` references `input` via `for=<ID_REF>` HTML technique, it's not hidden anymore. All  modern screen readers announce those labels without any problems.

<pre><code>
axe-core version: 4.3.2
</code></pre>

**Comment by WilcoFiers** [comment]
@Vittly have you tested this with assistive technologies? I know this is something we tested at the time. I can certainly give it another pass.

**Comment by Vittly** [comment]
> have you tested this with assistive technologies?

Sure. iOS VoiceOver, Android TalkBack and MacOS VoiceOver all works well announcing label if it's `aria-hidden`

**Comment by scottaohara** [comment]
Firefox will not report an accessible name for an input if you set it's label to `aria-hidden=true`.  Chromium browsers will still expose the label's content as the input's accessible name.

https://codepen.io/scottohara/pen/qBXGepO

This does _not_ work with Safari on macOS, or using Safari tech preview 134 (15.4) or Safari 15.0 with macOS 11.6.  Nor does it work with Safari on iOS 15.1.   So would be curious to see how you're getting different results, @Vittly - unless you are specifically testing with Chrome and VoiceOver?

**Comment by Vittly** [comment]
Actually @scottaohara you're right Safari on macOS ignores `label[aria-hidden]` as well as my iPhone 7, but Chrome 96.0.4664.55 (Official Build) on macOS announces it, so does TalkBack + Chrome on Pixel phone. Probably old test iPhone 5 with older iOS also announced it, but it's at the office and I can't check.
Seems like it can be freely closed. Sorry for false alarm
